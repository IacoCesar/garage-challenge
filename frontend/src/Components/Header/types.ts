export interface IHeader {
  testID?: string
  isHomeScreen?: boolean
  title?: string
  callHandleBack?: (navigation: any) => void
  callHandleChangeVisualMode?: (themeSelector: string) => void
}

export interface IProps extends IHeader {
  themeSelector: string
}
