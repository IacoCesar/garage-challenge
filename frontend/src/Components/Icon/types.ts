export interface IProps {
  testID?: string
  size: number
  name: string
  color: string
  type?: string
}
