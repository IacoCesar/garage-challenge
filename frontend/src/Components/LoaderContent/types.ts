export interface IProps {
  testID?: string
  size: number
  color: string
}
