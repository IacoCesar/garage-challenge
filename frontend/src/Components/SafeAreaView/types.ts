export interface IProps {
  testID?: string
  children: JSX.Element
}

export interface IThemeProps {
  themeSelector: string
}
