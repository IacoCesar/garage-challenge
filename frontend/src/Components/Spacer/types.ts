export interface IProps {
  testID?: string
  min: number
  max: number
}
