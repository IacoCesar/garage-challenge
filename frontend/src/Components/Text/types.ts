export interface ITextProps {
  color?: string
  fontSize?: number
  isBold?: boolean
}

export interface IProps extends ITextProps {
  label: string
  testID?: string
}
