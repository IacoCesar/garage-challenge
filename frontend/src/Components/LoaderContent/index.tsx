import React from 'react'
import LoadingIndicator from '@/Components/Loader'
import { IProps } from './types'
import S from './styles'
import { assignTestId } from '@/Helpers/TestId'

const Component: React.FC<IProps> = ({
  testID = 'LoaderContent',
  size,
  color
}) => {
  return (
    <S.Content {...assignTestId('View', testID)}>
      <LoadingIndicator
        testID={`${testID}_Content_LoadingIndicator`}
        size={size}
        color={color}
      />
    </S.Content>
  )
}

export default Component
