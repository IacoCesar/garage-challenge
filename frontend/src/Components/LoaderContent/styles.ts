import styled from 'styled-components/native'

const Content = styled.View`
  display: flex;
  width: 100%;
  height: 100%;
  justify-content: center;
  align-items: center;
  position: absolute;
  background-color: rgba(0, 0, 0, 0.3);
  z-index: 5;
`

export default {
  Content
}
