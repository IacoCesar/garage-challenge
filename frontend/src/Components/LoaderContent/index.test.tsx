import React from 'react'
import { IProps } from './types'
import { mockTestId } from '@/Helpers/TestId'
import LoaderContent from '.'
import { render } from '@/Helpers/TestHelpers'

describe('LoaderContent', () => {
  let props: IProps = {
    testID: 'LoaderContent',
    size: 23,
    color: 'white'
  }

  const loaderElement = mockTestId('View', `${props.testID!}`).testID

  it(`should render ${loaderElement}`, async () => {
    const wrapper = render(<LoaderContent {...props} />)

    const el_container = wrapper.getByTestId(loaderElement)
    expect(el_container).toBeTruthy()
  })

  it('should render snapshot', async () => {
    const wrapper = render(<LoaderContent {...props} />)

    expect(wrapper.toJSON()).toMatchSnapshot()
  })
})
