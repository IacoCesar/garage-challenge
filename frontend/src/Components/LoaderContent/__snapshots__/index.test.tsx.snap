// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoaderContent should render snapshot 1`] = `
<View
  accessible={false}
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "rgba(0, 0, 0, 0.3)",
        "display": "flex",
        "height": "100%",
        "justifyContent": "center",
        "position": "absolute",
        "width": "100%",
        "zIndex": 5,
      },
    ]
  }
  testID="View_LoaderContent"
>
  <ActivityIndicator
    accessible={false}
    animating={true}
    color="white"
    hidesWhenStopped={true}
    size={23}
    style={
      Array [
        Object {},
      ]
    }
    testID="ActivityIndicator_LoaderContent_Content_LoadingIndicator"
  />
</View>
`;
